### PR TITLE
Add WorkDoneProgress notification for service template generator

### LINKED ISSUE #34347: Service templates are not  provided for the initial completion request
**Description:**
We recently introduced the service template generation feature which looks up the listener symbols in the package, imported packages and distribution and generates dynamic service template completion items. The service item generation is done inn a lazy manner as the distribution look up would take around 10 seconds to complete. Hence, The initialization of the service template generator  is done at the initial completion request   to the ModulePartNodeContext completion provider at the cursor postion depcited below.[1]

```ballerina
public function main() {

}

service <cursor>
```

[1] https://github.com/ballerina-platform/ballerina-lang/blob/04f2c2ea37880970f26bb33ce15c10d7af1f4a8e/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ModulePartNodeContext.java#L163

At this initialization, we asynchronously load the visible listener symbols from the ballerina distribution and the completion request progresses without waiting for the listener symbols to be found. Hence, at the initial request user might not get only the default snippets  as depicted below.
initial

When the said async process completes the completion list for the context is as follows. 
second

To fill the gap in the user experience we can use a WorkDoneProgress notification that indicates the user that the symbol loading is complete.

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$subject

Fixes #34347 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/ServiceTemplateGenerator.java [hunk 1]
```diff
@@ -89,7 +98,34 @@ public boolean initialized() {
...
+                // Initialize progress notification
+                WorkDoneProgressCreateParams workDoneProgressCreateParams = new WorkDoneProgressCreateParams();
+                workDoneProgressCreateParams.setToken(taskId);
+                languageClient.createProgress(workDoneProgressCreateParams);
+
+                // Start progress
+                WorkDoneProgressBegin beginNotification = new WorkDoneProgressBegin();
+                beginNotification.setTitle(TITLE_INITIALIZE);
+                beginNotification.setCancellable(false);
+                beginNotification.setMessage("Loading listener symbols from the distribution");
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/ServiceTemplateGenerator.java:116]: Final message shown will be `Initialize Service Template Generator: Loading listener symbols from the distribution` right? If yes, it's a bit lengthy. How about `Service Template Generator: Initializing...` where title is `Service Template Generator`. WDYT?

**Review by IMS94 — CHANGES_REQUESTED** [review]

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/ServiceTemplateGenerator.java (hunk 1 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/ServiceTemplateGenerator.java:116]: +1

Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/util/ServiceTemplateGenerator.java]: Done

**Review by IMS94 — APPROVED** [review]
LGTM!

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Pull request merged**